app/uk/gov/hmrc/economiccrimelevyregistration/services/RegistrationValidationService.scala

```diff
@@ -38,19 +39,28 @@ class RegistrationValidationService @Inject() (clock: Clock, schemaValidator: Sc
 
   type ValidationResult[A] = ValidatedNel[DataValidationError, A]
 
-  def validateRegistration(registration: Registration): ValidationResult[EclSubscription] =
-    transformToEclSubscription(registration) match {
-      case Valid(eclSubscription) =>
-        schemaValidator
-          .validateAgainstJsonSchema(
-            eclSubscription.subscription,
-            SchemaLoader.loadSchema("create-ecl-subscription-request.json")
-          )
-          .map(_ => eclSubscription)
-      case invalid                => invalid
+  def validateRegistration(registration: Registration): ValidationResult[Either[EclSubscription, Registration]] =
+    registration.entityType match {
+      case Some(Other) => validateOtherEntity(registration)
+      case _           =>
+        transformToEclSubscription(registration) match {
+          case Valid(either) =>
+            either match {
+              case Left(eclSubscription) =>
+                schemaValidator
+                  .validateAgainstJsonSchema(
+                    eclSubscription.subscription,
+                    SchemaLoader.loadSchema("create-ecl-subscription-request.json")
+                  )
+                  .map(_ => either)
```

Comment:
A couple of things:

- You don't need nested matching like this, you can match immediately on the valid like:
`case Valid(Left(eclSubscription)) => ...` and `case Valid(Right(registration)) => ...` 
- This is not exhaustive, you are not handling the `Right(registration)` which results in a compiler warning, although not technically possible based on the implementation, we should satisfy the compiler by handling the right such as:
`case Valid(Right(_)) => DataValidationError(DataInvalid, "Data was not transformed into a valid ECL subscription").invalidNel`

Reply:
Refactored code.